### DISCUSSION TITLE
How do I debug V8 itself?

### DISCUSSION THREAD

**Discussion by AnsisMalins** [ClearFoundry/ClearScript#619] [Q&A]
What do I do to have V8 itself (e.g. api.cc) open in Visual Studio, or any other editor, with syntax highlighting, navigation, breakpoints, and so on?

**Comment by ClearScriptLib** [comment]
Hi @AnsisMalins,

You can build ClearScript in Debug mode (starting with "V8Update debug") and set your .NET project to debug both native and managed code via the project properties in Visual Studio. Once the debugger is attached, you should then be able to step from managed code into native code and vice versa.

Also, if you've run V8Update, you should be able to open the V8 source directory in Visual Studio: "devenv _[ClearScriptRoot]_\V8\build\v8".

Unfortunately, V8 is a bit of a nightmare scenario for Visual Studio. It's extremely large, there are a ton of macros, a lot of code is inlined even in Debug mode, and of course there's the JIT compiler (unless you disable it).

Good luck!
